src/data-table/ColumnSelectorDialog.tsx

```diff
@@ -24,12 +26,33 @@ export function ColumnSelectorDialog<T extends ReferenceObject>(
         onChange,
         onCancel,
         allowReorderingColumns = true,
+        keepDisabledColumns = true,
     } = props;
-    const sortableColumns = columns.map(({ name, text: label, disabled }) => ({
-        label,
-        value: name,
-        disabled: disabled ?? false,
-    }));
+    const sortableColumns = columns.map(
+        ({ name, text: label, disabled }): TransferOption => ({
+            label,
+            value: name.toString(),
+            disabled: disabled ?? false,
+        })
+    );
+
+    const selectedColumns = React.useMemo((): string[] => {
+        const disableColumns = columns.filter(col => col.disabled).map(col => col.name.toString());
+        return _(visibleColumns).map(String).concat(disableColumns).uniq().value();
+    }, [visibleColumns, columns]);
+
+    const updateSelectedColumns = React.useCallback(
+        ({ selected }: { selected: Array<keyof T> }) => {
```

Comment:
Nice to have this immutable.

[minor] It's more idiomatic `useCallback<T>`, being T a function of the prop type (and using NonNullable, if it's optional).